### PR TITLE
Add docs index to README.md

### PR DESCRIPTION
**Pull request by mik-laj**
I delete the link to the IMAGES.rst file because it describes a lot of details relevant to contributors, but for users, we have separate documentation.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
I believe README.md is for developers not users. IMAGES.rst is quite relevant here.

**Comment by potiuk** [comment]
The user's documentation is at  https://airflow.apache.org/docs/

**Comment by mik-laj** [comment]
> I believe README.md is for developers not users. IMAGES.rst is quite relevant here.

Not only. Some sections are for developers and others are for users.  This section seems to be intended for users who are getting started with Airflow, so the documentation on CI images is not helpful for them.

**Comment by mik-laj** [comment]
I added.back the link to IMAGES.rst. It is in a section indented for contributor now.

File: README.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ through a more complete [tutorial](https://airflow.apache.org/docs/apache-airflo
 For more information on Airflow Improvement Proposals (AIPs), visit
 the [Airflow Wiki](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals).
 
-Official Docker (container) images for Apache Airflow are described in [IMAGES.rst](https://github.com/apache/airflow/blob/main/IMAGES.rst).
+Documentation for dependent projects like provider packages, Docker image, Helm Chart, you'll find it in [the documentation index](http://airflow.apache.org/docs/).
```
Comment (mik-laj) [inline README.md]:
```suggestion
Documentation for dependent projects like provider packages, Docker image, Helm Chart, you'll find it in [the documentation index](https://airflow.apache.org/docs/).
```

**Review by potiuk — APPROVED** [review]

**Pull request merged**